A few quick thread fixes

liewegas opened the pull request.

liewegas committed:
updateThreadStatusFromMessage: use both thread ts and channel for update

benweissmann commented on src/db_api_util.ts:
```diff
@@ -557,14 +557,26 @@ export async function logThreadToDb(
+    !databaseMessageEntry.slackChannel
+  ) {
+    // no thread info to update
+    return;
```
When would this case happen? Should we be logging a warning here? It sorta feels like this check should be in whatever's calling `updateThreadStatusFromMessage` so if these fields are unexpectedly missing, we don't silently ignore that.

liewegas replied:
Repushed with the check in teh caller.  The errors were happening during the initial welcome and first message of the 'pick your state' exchange, before we have a slack thread.

liewegas committed:
fix router test

liewegas committed:
do not adjust thread status if message is not part of a slack thread

This avoids errors during the initial welcome and (early) 'pick your state'
exchange.
